datadog.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// Package datadog contains a Datadog exporter.
+package datadog
+
+import (
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	defaultHost   = "localhost"
+	defaultPort   = "8125"
+	opencensusTag = "source:Opencensus"
+)
+
+var (
+	exporter *Exporter
+	tags     = []string{opencensusTag}
+	reg      = regexp.MustCompile("[^a-zA-Z0-9]+")
+)
+
+// Exporter exports stats to Datadog.
+type Exporter struct {
+	opts      Options
+	collector *collector
+}
+
+// ExportView exports to Datadog if view data has one or more rows.
+func (e *Exporter) ExportView(vd *view.Data) {
+	if len(vd.Rows) == 0 {
+		return
+	}
+	e.collector.addViewData(vd)
+}
+
+// Options contains options for configuring the exporter.
+type Options struct {
+	// Namespace specifies the namespace to which metrics are appended.
+	Namespace string
+
+	// Host for DogStatsD connection
+	Host string
```

Comment:
I understand your motivations, I'd argue further that it'd be nice to have just one string field for the following reasons:
* We can see countless examples in the standard library where this is a pattern in Go, such as: [`http.ListenAndServe`](https://golang.org/pkg/net/http/#ListenAndServe) and it's sister methods, [`net.Dial`](https://golang.org/pkg/net/#Dial) among with other dialers, etc.
* It is what `statsd.New` expects as a parameter
* We will also have another field in this structure: the address for the tracing agent. It'll be much nicer to have two fields (let's say `StatsAddr` and `TraceAddr` rather than four: `StatsHost`, `StatsPort`, `TraceHost`, `TracePort`)
* From a user perspective it's much easier to type `{StatsAddr: "host:port"}` than it is `{StatsHost: "host", StatsPort: "port"}`
* No need for "gluing" logic like `getEndpoint`

If you and your team still feel strongly about having it in this form, I'm happy for you to keep it, I just wanted to share my motivations and perhaps you can consider them.